### DISCUSSION TITLE
Forward double slashes to prevent path expansion doesn't work with IPFS cli util

### DISCUSSION THREAD

**Discussion by ftoh** [git-for-windows/git#3024] [Q&A]
Running go-ipfs produce error
```shell
λ ipfs cat /ipfs/QmQPeNsJPyVWPFDVHb77w8G42Fvo15z4bG2X8D2GhfbSXc/readme                                          
Error: invalid path "C:/Users/admin/scoop/apps/cmder-full/current/vendor/git-for-windows/ipfs/QmQPeNsJPyVWPFDVHb
77w8G42Fvo15z4bG2X8D2GhfbSXc/readme": illegal base32 data at input byte 0                                       
```

I found a double slash solution very quickly, but it works differently for a path from multiple folders.
```shell
λ node -p process.argv //test
[ 'C:\\Users\\admin\\scoop\\apps\\nodejs\\current\\node.exe', '/test' ]

λ node -p process.argv //test/test
[
  'C:\\Users\\admin\\scoop\\apps\\nodejs\\current\\node.exe',
  '//test/test'
]
```

**I'm pretty sure the double slash behavior should be the same in all cases.**

It took me some time to find information about MSYS_NO_PATHCONV=1 and possible solution to run ipfs
`alias ipfs='MSYS_NO_PATHCONV=1 ipfs'`

<details>
  <summary>Soft version</summary>

 -  **Which version of Git for Windows are you using? Is it 32-bit or 64-bit?**

```
$ git --version --build-options

git version 2.29.1.windows.1
cpu: x86_64
built from commit: e6d53c456169235fe72602e2b4d985934fa9ad23
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
```

 - **Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?**
 - 10 20H2 64-bit

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19042.746]
```
</details>

**Comment by rimrul** [comment]
The `//` behaviour is intended for windows commands that don't accept command line options starting with a dash. Those generally don't contain a second slash. Because of that, when the path conversion code sees a parameter starting with two slashes and containing one or more slashes after that it concludes you're not passing windows-style command line options or a unix-style path, so it does no conversion.

Why are you trying to pass an absolute unix path to a native windows application anyways?

**Comment by ftoh** [comment]
> Why are you trying to pass an absolute unix path to a native windows application anyways?

This is how IPFS addresses look like. See: https://docs.ipfs.io/concepts/what-is-ipfs/#content-addressing

Another example where escaping unix paths can be useful is docker (Hyper-V, WSL2 or legacy ToolBox).
```shell
$ cd /myserver
$ echo "hello world" > index.html
$ docker run -d -p 80:80 \
    -v /myserver/index.html:/usr/share/caddy/index.html \
    caddy
...
$ curl http://localhost/
hello world
```
While one (`/myserver/index.html`) expands in the absolute windows path, and the second (`/usr/share/caddy/index.html`) should remain unchanged, where the slash escaping will help.

**Comment by dscho** [comment]
> It took me some time to find information about MSYS_NO_PATHCONV=1 and possible solution to run ipfs
> `alias ipfs='MSYS_NO_PATHCONV=1 ipfs'`

That is indeed the best way to handle this issue.
